### PR TITLE
Remove backend domain model getters and setters

### LINKED ISSUE #94: Remove redundant getters and setters
**Is your feature request related to a problem? Please describe.**
There are some getters and setters method in the backend currently doing nothing, as properties in a class can be retrieved by `Class.Property` without having to call `Class.getProperty()`

**Describe the solution you'd like**
Remove the redundant getters and setters

### Feature Approval:
- [x]  Feature approved by: @SreenikethRaghavan

### PR DESCRIPTION
**Pull request by SreenikethRaghavan**
This PR fixes #94

Removed getter and setter functions from domain classes. As JavaScript properties can be accessed directly, the getter and setter functions for domain properties are redundant. Therefore, these functions have been removed and their usage has been replaced with directly accessing the properties.

# Type of change:
* Refactor

# Checklist ✔
* Have you merged main into your branch?
* Have you tested your changes to ensure it works as expected and does not break existing functionality?
* If applicable, please ensure sufficient tests are added that is related to the changes.
* Please ensure this PR has a `label`, is linked to an `issue` and is related to a `project`
* If new documentation is required for this change, have you created a new documentation issue that describes the documentation needed?

### CODE REVIEW

File: backend/domain/Room.js [hunk 1]
```diff
@@ -21,22 +21,10 @@ class Room {
...
 
-    getRoomID() {
-        return this.roomID;
-    }
-
-    getGameState() {
-        return this.gameState;
-    }
-
     setGameState(gameState) {
```
Comment (parfei) [inline backend/domain/Room.js:24]: Can we remove this setter as well please?

File: backend/domain/Room.js [hunk 2]
```diff
@@ -74,20 +62,12 @@ class Room {
...
             this.players.push(player);
         }
     }
 
-    getRoundNumber() {
-        return this.roundNumber;
-    }
-
     incrementRoundNumber() {
         this.roundNumber++;
```
Comment (parfei) [inline backend/domain/Room.js:72]: I am pretty sure ESLint will convert `++` into `+= 1` but @MKTPR's hasn't been merged in, so might be worth just changing it here anyway

**Review by parfei — CHANGES_REQUESTED** [review]
Nice work Sree :) 👍 ! This will make our domain models nicer to work with.

**Review by jteo97 — APPROVED** [review]

**Comment by parfei** [comment]
Also a sidenote: if we are prefixing refactor branches with `refactor`, would you be able to add that to the `Contributing` markdown file?

File: backend/domain/Room.js (hunk 1 above)
Comment (SreenikethRaghavan) [reply backend/domain/Room.js:24]: I've now removed it

File: backend/domain/Room.js (hunk 2 above)
Comment (SreenikethRaghavan) [reply backend/domain/Room.js:72]: Ok yup, I can change that. Personally I don't see an issue with ++, but for consistency purposes will change it.

**Review by parfei — APPROVED** [review]
Nice work Sree! <3

**Review by seif-y — APPROVED** [review]
Good stuff Sree, looks ready to merge

**Pull request merged**